core/src/main/java/jenkins/util/InterceptingExecutorService.java

```diff
@@ -11,13 +12,18 @@
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 
 /**
  * {@link ExecutorService} that wraps all the tasks that run inside.
  *
  * @author Kohsuke Kawaguchi
  * @since 1.557
  */
+@Deprecated
+@Restricted(NoExternalUse.class)
```

Comment:
I would prefer DoNotUse.clasd which prevents even accidental use inside core (as per commentary this would pick up the classes that extend this)